### PR TITLE
[BEAM-6777] Add enable_health_checker flag for Dataflow FnAPI worker (Python)

### PR DESCRIPTION
**Pull request by robinyqiu**
The `enable_health_checker` flag will be added by default for Python jobs using Dataflow FnAPI worker (e.g. Python streaming on Dataflow). Dataflow will enable the channel between the health checker and the worker once it sees this flag.

R: @aaltay @angoenka 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by robinyqiu** [comment]
Run Portable_Python PreCommit

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 1]
```diff
@@ -177,20 +177,27 @@ def __init__(self, packages, options, environment_version, pipeline_url):
...
       # till version 2.4.
-      debug_options_experiments = self.debug_options.experiments
       if ('use_multiple_sdk_containers' not in debug_options_experiments and
           'no_use_multiple_sdk_containers' not in debug_options_experiments):
-        self.debug_options.experiments.append('use_multiple_sdk_containers')
+        debug_options_experiments.append('use_multiple_sdk_containers')
+      # Add enable_health_checker flag if it's not already present. Do not
+      # add the flag if 'disable_health_checker' is present.
+      # TODO: Cleanup enable_health_checker once we deprecate Python SDK
+      # till version 2.14.
```
Comment (angoenka) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Shall we also file a jira for the cleanup.

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py [hunk 2]
```diff
@@ -500,6 +500,34 @@ def test_experiment_use_multiple_sdk_containers(self):
...
+
+    pipeline_options = PipelineOptions(
+        ['--project', 'test_project', '--job_name', 'test_job_name',
+         '--temp_location', 'gs://test-location/temp',
+         '--experiments', 'beam_fn_api',
+         '--experiments', 'disable_health_checker'])
+    environment = apiclient.Environment(
+        [], pipeline_options, 1, FAKE_PIPELINE_URL)
+    self.assertNotIn(
+        "enable_health_checker", environment.proto.experiments)
```
Comment (angoenka) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: We can also add 
`    self.assertIn('disable_health_checker', environment.proto.experiments)`

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py [hunk 3]
```diff
@@ -500,6 +500,34 @@ def test_experiment_use_multiple_sdk_containers(self):
...
         "use_multiple_sdk_containers", environment.proto.experiments)
 
+  def test_experiment_enable_health_checker(self):
+    pipeline_options = PipelineOptions(
+        ['--project', 'test_project', '--job_name', 'test_job_name',
+         '--temp_location', 'gs://test-location/temp',
+         '--experiments', 'beam_fn_api'])
+    environment = apiclient.Environment(
+        [], pipeline_options, 1, FAKE_PIPELINE_URL)
+    self.assertIn("enable_health_checker", environment.proto.experiments)
```
Comment (angoenka) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: Please use single quotes in python.
`'enable_health_checker'`

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py (hunk 1 above)
Comment (robinyqiu) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Good suggestion! Thanks.

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py (hunk 3 above)
Comment (robinyqiu) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: Done. Thanks for paying attention to style issue.

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py (hunk 2 above)
Comment (robinyqiu) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: Done.

**Pull request merged**